src/Util/File.cc

```diff
@@ -86,9 +86,9 @@ CARTA::FileType GuessImageType(const std::string& path_string, bool check_conten
         fs::path path(path_string);
         auto filename = path.filename().string();
 
-        if (EndsWith(filename, ".fits") || EndsWith(filename, ".FITS") || EndsWith(filename, ".fz") || EndsWith(filename, ".fits.gz")) {
+        if (HasSuffix(filename, ".fits") || HasSuffix(filename, ".FITS") || HasSuffix(filename, ".fz") || HasSuffix(filename, ".fits.gz")) {
```

Comment:
1. It looks like you're using the case sensitive check everywhere.
2. If this becomes case-insensitive, one of the conditions will be redundant.